src/main/java/org/dependencytrack/model/CryptoCertificateProperties.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * This file is part of Dependency-Track.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.dependencytrack.model;
+
+import java.io.Serializable;
+import java.util.Date;
+
+import javax.jdo.annotations.Column;
+import javax.jdo.annotations.IdGeneratorStrategy;
+import javax.jdo.annotations.PersistenceCapable;
+import javax.jdo.annotations.Persistent;
+import javax.jdo.annotations.PrimaryKey;
+
+import org.dependencytrack.resources.v1.serializers.Iso8601DateSerializer;
+import org.dependencytrack.util.DateUtil;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@PersistenceCapable(table= "CERTIFICATE_PROPERTIES")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CryptoCertificateProperties implements Serializable {
+
+    private static final long serialVersionUID = 6421255046930674723L;
+
+    @PrimaryKey
+    @Persistent(valueStrategy = IdGeneratorStrategy.NATIVE)
+    @JsonIgnore
+    private long id;
+
+
+    /*
+     * certificateProperties
+     */
+    @Persistent
+    @Column(name = "SUBJECT_NAME", jdbcType = "VARCHAR", length=255)
+    private String subjectName;
+    @Persistent
+    @Column(name = "ISSUER_NAME", jdbcType = "VARCHAR", length=255)
+    private String issuerName;
+    @Persistent
+    @Column(name = "NOT_VALID_BEFORE", jdbcType = "TIMESTAMP")
+    @JsonSerialize(using = Iso8601DateSerializer.class)
+    private Date notValidBefore;
+    @Persistent
+    @Column(name = "NOT_VALID_AFTER", jdbcType = "TIMESTAMP")
+    @JsonSerialize(using = Iso8601DateSerializer.class)
+    private Date notValidAfter;
+    @Persistent
+    @Column(name = "SIGNATURE_ALGORITHM_REF", jdbcType = "VARCHAR", length=64)
+    private String signatureAlgorithmRef;
+    @Persistent
+    @Column(name = "SUBJECT_PUBLIC_KEY_REF", jdbcType = "VARCHAR", length=64)
+    private String subjectPublicKeyRef;
```

Comment:
You're right, they should be resolved. I haven't done this yet because the issue appears in many places where the standard has xxxRef data fields. In most cases, these are bomrefs that could be resolved to components. Most prominently, the bomref resolution question arises in the dependencies. Here DT currently stores a JSON structure which imho cannot not be the final solution. Hence I thought that bomref resolution ist still an open issue and left the refs plain for the time being.

Reply:
In principle it's the same tree structure. The bomrefs in the ref fields must obviously point to other components in the BOM otherwise they would be dangling. In a sense, the standard is "over-specified": You can have ref fields that point to components **and** you can have `dependency` objects that also state that one component depends on another. It is not clear if the specification of a `signatureAlgorithmRef` requires the specification of a dependency object.

I think our final goal should be to model `signatureAlgorithmRef` as 
```java
 private Component subjectPublicKeyRef;
```
that is set as the result of the bomref resolution. Maybe we can make these fields transient because the get re-created on the fly when the BOM is loaded.